lib/config/presets/index.ts

```diff
@@ -218,15 +219,22 @@ export async function getPreset(
   }
   const { presetSource, repo, presetPath, presetName, tag, params } =
     parsePreset(preset);
-  let presetConfig = await presetSources[presetSource].getPreset({
-    repo,
-    presetPath,
-    presetName,
-    tag,
-  });
+  const cacheKey = `preset:${preset}`;
+  const cachedResult = memCache.get<Preset>(cacheKey);
+  let presetConfig = is.undefined(cachedResult)
+    ? await presetSources[presetSource].getPreset({
+        repo,
+        presetPath,
+        presetName,
+        tag,
+      })
+    : cachedResult;
```

Comment:
Still don't like that syntax, hard to read
```suggestion
  let presetConfig = memCache.get<Preset>(cacheKey);
  if (is.undefined(cachedResult)) {
    presetConfig = await presetSources[presetSource].getPreset({
        repo,
        presetPath,
        presetName,
        tag,
      });
    }
```

Let's use a little more verbose syntax.